src/main/java/com/hcl/appscan/jenkins/plugin/scanModes/TestOnly.java

```diff
@@ -0,0 +1,118 @@
+        m_userNameTestScan = accessIdTestScan;
+        m_passwordTestScan = Secret.fromString(secretKeyTestScan);
+        m_exploreDataTestScan = exploreDataTestScan;
+    }
+
+    @DataBoundSetter
+    public void setLoginTypeTestScan(String loginTypeTestScan) {
+        m_loginTypeTestScan = loginTypeTestScan;
+    }
+
+    public String getLoginTypeTestScan() {
+        return m_loginTypeTestScan;
+    }
+
+    @DataBoundSetter
+    public void setAccessIdTestScan(String userNameTestScan) {
+            m_userNameTestScan = userNameTestScan;
+    }
+
+    public String getAccessIdTestScan() {
+        return m_userNameTestScan;
+    }
+
+    @DataBoundSetter
+    public void setSecretKeyTestScan(String passwordTestScan) {
+            m_passwordTestScan = Secret.fromString(passwordTestScan);
+    }
+
+    public String getSecretKeyTestScan() {
+        return Secret.toString(m_passwordTestScan);
+    }
+
+    @DataBoundSetter
+    public void setTrafficFileTestScan(String trafficFileTestScan) {
+            m_trafficFileTestScan = trafficFileTestScan;
+    }
+
+    public String getTrafficFileTestScan() {
+        return m_trafficFileTestScan;
+    }
+
+    @DataBoundSetter
+    public void setExploreDataTestScan(String exploreDataTestScan) {
+            m_exploreDataTestScan = exploreDataTestScan;
+    }
+
+    public String getExploreDataTestScan() {
+        return m_exploreDataTestScan;
+    }
+
+    public String isLoginTypeTestScan(String loginTypeName) {
+        if (m_loginTypeTestScan != null) {
+            return m_loginTypeTestScan.equalsIgnoreCase(loginTypeName) ? "true" : "";
+        } else if (loginTypeName.equals("Manual")) { //Default
+            return "true";
+        }
+        return "";
+    }
+
+    @Override
+    public Map<String, String> configureScanProperties(Map<String, String> properties, VariableResolver<String> resolver) {
+        properties.put("scanType", TEST_ONLY);
+        properties.put("loginType", m_loginTypeTestScan);
+        properties.put("exploreData", resolver == null ? m_exploreDataTestScan : resolvePath(m_exploreDataTestScan, resolver));
+        if ("Manual".equals(m_loginTypeTestScan)) {
+            properties.put("trafficFile", resolver == null ? m_trafficFileTestScan : resolvePath(m_trafficFileTestScan, resolver));
+        } else if("Automatic".equals(m_loginTypeTestScan)) {
+            properties.put("userName", resolver == null ? m_userNameTestScan : Util.replaceMacro(m_userNameTestScan, resolver));
+            properties.put("password", resolver == null ? Secret.toString(m_passwordTestScan) : Util.replaceMacro(Secret.toString(m_passwordTestScan), resolver));
+        }
+        return properties;
+    }
+
+    @Symbol("test_only") //$NON-NLS-1$
+    @Extension
+    public static final class DescriptorImpl extends ScanModeDescriptor {
+
+        @Override
+        public String getDisplayName() {
+            return "Test Only";
```

Comment:
Can you use the TEST_ONLY constant defined above here?

Reply:
Okay, sure.